concepts/Vulnerability_disclosure_and_embargo_policy/00004-Vulnerability_disclosure_and_embargo_policy.md

```diff
@@ -0,0 +1,221 @@
+we welcome their efforts and how we will manage their disclosure. They must
+also be able to have an idea of how we use security embargoes.
+
+In other words:
+- We want:
+  - to encourage reporting of security issues (ie: “attack us!”)
+  - to make it easy to report security issues
+  - to maintain and extend our policy of safe defaults / discouraging people
+  from using bad security practices
+  - to recognise and show appreciation for the time spent researching and
+  reporting
+  - to make it possible for people who may be at risk to report issues
+  pseudonymously
+
+- We need:
+  - a clear policy, describing both:
+    - how to report; and
+    - what to expect during the process
+  - a main reporting channel:
+    - secondary channels, should the main one fail/not be available/be
+  overloaded.
+  - a list of people who can be reached directly
+  - a place to list security reports and give attribution
+
+
+## Tutorial
+
+This RFC aims to provide a set of documents (wiki pages) that describe the 
+Enarx vulnerability disclosure policy and the list of security advisories and
+ fixes.
+
+The two pages are "Vulnerability Reporting and Embargo Policy" and "Enarx 
+Security Advisories"
+
+--- 
+### Vulnerability Reporting and Embargo Policy
+The Enarx project aims to be as safe to use as possible and therefore to 
+produce code that is as secure as possible. As such, we welcome any and all 
+research into Enarx and appreciate your efforts to take the time to 
+responsibly disclose any issues you find.
+
+Vulnerabilities may be discovered in:
+1. Enarx code
+2. Enarx dependencies (including compilers)
+3. Enarx documentation
+4. 3rd party hardware or firmware
+
+All security bugs in Enarx should be reported by email to security@enarx.io. 
+This list is delivered to a small security team. Your email will be 
+acknowledged within 48 hours, and you’ll receive a more detailed response to 
+your email within 96 hours indicating the next steps in handling your report.
+ If you wish, you may encrypt your report using our public key. This key is 
+also on keys.openpgp.org and reproduced below.
+
+Be sure to use a descriptive subject line to avoid having your report be 
+missed (as with any publicly listed addresses, it is expected this one will 
+get a lot of spam). After the initial reply to your report, the security team
+ will endeavor to keep you informed of the progress being made towards a fix 
+and full announcement. As recommended by 
+[RFPolicy](https://dl.packetstormsecurity.net/papers/general/rfpolicy-2.0.txt),
+ these updates will be sent at least every five working days.
+
+If you have not received a reply to your email within 72 hours, or have not 
+heard from the security team for the past five days, there are a few steps you
+ can take (in order):
+- Contact the current security coordinator (Nathaniel McCallum (public key)) directly.
+- Contact the back-up contact (Mike Bursell (public key)) directly.
```

Comment:
We have to decide if we're going to maintain public keys.  Neither @npmccallum nor I currently does.

Reply:
This is one of the big questions this document raises.

I'm warming up more and more to the idea of treating email like  public channel, and using it (or the chat) to set up more secure communication channels. The question then becomes: what communication channel do we set up?

Reply:
Does Zulip have a way of managing this, by any chance?

Reply:
I'm not sure if this is exactly what you're looking for, but Zulip supports [sending an email to a (private?) stream](https://zulipchat.com/help/message-a-stream-by-email).

I am not sure if it's bidirectional though, so if the Enarx security council maintains a private stream and they receive a vulnerability disclosure, I am not sure if their responses into the stream will be relayed back by email to the security researcher.

I don't have enough data to know if Zulip is an appropriate place for this in terms of security and privacy.

Reply:
That's a neat Zulip function, thanks for pointing it out. However documentation says nothing about PGP-encrypted email. This means that the email would be unencrypted until it reaches the Zulip server, at which point it's private (but not E2E-encrypted, as the Zulip server is able to relay the content of the email clearly to the stream). Furthermore, replies to the stream to the initial sender would also be unencrypted, as the Zulip server would not know which public PGP key to encrypt it to. So while cool, this is not really an option for us i fear.

Reply:
Suggest we close the Zulip thread here, as we've gone with Rocket.